sequencers/single/sequencer.go

```diff
@@ -5,20 +5,54 @@ import (
 	"context"
 	"errors"
 	"fmt"
+	"strings"
+	"sync"
 	"time"
 
 	ds "github.com/ipfs/go-datastore"
 	"github.com/rs/zerolog"
 
 	coreda "github.com/evstack/ev-node/core/da"
 	coresequencer "github.com/evstack/ev-node/core/sequencer"
+	"github.com/evstack/ev-node/pkg/genesis"
 )
 
-// ErrInvalidId is returned when the chain id is invalid
 var (
+	// ErrInvalidId is returned when the chain id is invalid
 	ErrInvalidId = errors.New("invalid chain id")
+	// ErrForceInclusionNotConfigured is returned when forced inclusion is not configured
+	ErrForceInclusionNotConfigured = errors.New("forced inclusion namespace not configured")
 )
 
+// ForcedInclusionEvent represents forced inclusion transactions retrieved from DA
+type ForcedInclusionEvent = struct {
+	Txs           [][]byte
+	StartDaHeight uint64
+	EndDaHeight   uint64
+}
+
+// DARetriever defines the interface for retrieving forced inclusion transactions from DA
+type DARetriever interface {
+	RetrieveForcedIncludedTxsFromDA(ctx context.Context, daHeight uint64) (*ForcedInclusionEvent, error)
+}
+
+// DARetrieverAdapter adapts any retriever that returns a compatible event type
+type DARetrieverAdapter struct {
```

Comment:
Why do you need the adapter? The `block.DARetriever` implements the `DARetriever` interface and can be passed to the constructor directly.